File: spec/python/test_process_xor_const.py
import unittest

from process_xor_const import ProcessXorConst

class TestProcessXorConst(unittest.TestCase):
    def test_process_xor_const(self):
        r = ProcessXorConst.from_file("src/process_xor_1.bin")

        self.assertEqual(r.key, 0xff)
        self.assertEqual(r.buf, b"foo bar")

File: spec/lua/test_process_xor_const.lua
local luaunit = require("luaunit")

require("process_xor_const")

TestProcessXorConst = {}

function TestProcessXorConst:test_process_xor_const()
    local r = ProcessXorConst:from_file("src/process_xor_1.bin")

    luaunit.assertEquals(r.key, 0xff)
    luaunit.assertEquals(r.buf, "foo bar")
end
